### DISCUSSION TITLE
Efficient representation of sparse Pauli error

### DISCUSSION THREAD

**Discussion by EmilianoG-byte** [tencent-quantum-lab/tensorcircuit#205] [Q&A]
Hi (again),

Is there a way to efficiently represent/implement a sparse Pauli channel? For instance, in Qiskit I could use the [`pauli_error` function](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.noise.pauli_error.html) with a [Pauli String representation](https://docs.quantum.ibm.com/api/qiskit/qiskit.quantum_info.Pauli) as input.

This "efficient" representation is particularly useful, because I am considering up to weight-2 Pauli strings, so for a high number of qubits `n` not having to save the $4^n$ vector would be useful. I am aware of the [general_depolarizing_channel](https://tensorcircuit.readthedocs.io/en/latest/api/channels.html?highlight=amplitude%20damping#tensorcircuit.channels.generaldepolarizingchannel) in TC, but this would imply running into the latter problem.

Any idea?

**Comment by EmilianoG-byte** [comment]
@refraction-ray any idea? 🤔

**Comment by refraction-ray** [comment]
What you mean is Kraus operator like $[X_1X_2, X_1Y_2, ..., Z_2Y_3, ... Z_{n-1}Z_n]$? If applied in non-jit version, it can be done by Monte Carlo trajectory simulation, where one of the local operators is applied on the system depending on the random number.

However, if jit is enabled, I cannot think of an efficient way to implement this for now. The essential problem is the same as applying two-qubit gates on random indices while maintaining jit, see related discussion: https://github.com/tencent-quantum-lab/tensorcircuit/issues/171#issuecomment-1666696484

**Reply by refraction-ray** [reply]
Ah, actually if the two-qubit operators are all local and Pauli, then an efficient jittable MPO representaion could exist. It is more like this case: https://github.com/tencent-quantum-lab/tensorcircuit/issues/171#issuecomment-1666640861. Therefore, you can run jittable efficient simulation with Monte Carlo trajectory approach.

**Reply by EmilianoG-byte** [reply]
Actually I don’t necessarily need to keep it jittable for now. So there is no way of doing this with a DMCircuit? 

My problem at the moment is that when going beyond 6 qubits, the list of Kraus of operators is so long that even just applying the noise with `circuit_with_noise` becomes super slow. I thought I could avoid this by applying the local noise to specific indices instead of creating the sparse tensor on the full system. I see the link you sent attempts this with MonteCarlo, right?

**Reply by EmilianoG-byte** [reply]
Oh no wait, the solution is using MPO tensors right? Let me try it out!!
